AdblockPlusSafari/FilterList.m

```diff
@@ -45,16 +45,24 @@ + (NSArray *)boolProperties
 
 - (instancetype)initWithDictionary:(NSDictionary *)dictionary;
 {
+  if (!dictionary) {
+    return nil;
+  }
+
   if (self = [super init]) {
     for (NSString *key in [FilterList allProperties]) {
       [self setValue:[dictionary valueForKey:key] forKey:key];
     }
   }
+
   return self;
 }
 
 - (NSDictionary *)dictionary
 {
+  NSAssert(self.fileName, @"Filename must be set!");
```

Comment:
This should not happen. FilterList is create only from special dictionary, and that dictionary is not accessed from outside.